### ISSUE TITLE
Dart_TypedData_kUint64 error: unimplemented code

### ISSUE THREAD

**Issue by vaind** [dart-lang/sdk#44384]
* Dart SDK version: 2.10.4 (stable) (Wed Nov 11 13:35:58 2020 +0100) on "linux_x64"
Trying to send async callback data to native port using `Dart_PostCObject_DL()` - an `Dart_CObject_kArray` of dynamic values (one is Int64 `Dart_CObject_kInt64` followed by two Uint64List `Dart_TypedData_kUint64`) fails during dart runtime:

```
./../runtime/vm/dart_api_message.cc: 1086: error: unimplemented code
version=2.10.4 (stable) (Wed Nov 11 13:35:58 2020 +0100) on "linux_x64"
pid=946331, thread=946354, isolate_group=main(0x558ffa12f700), isolate=main(0x558ffa392000)
isolate_instructions=558ff75fcb80, vm_instructions=558ff75fcb80
  pc 0x0000558ff788cbcc fp 0x00007f076e87cc10 dart::Profiler::DumpStackTrace(void*)+0x7c
  pc 0x0000558ff75fccb2 fp 0x00007f076e87ccf0 dart::Assert::Fail(char const*, ...)+0x82
  pc 0x0000558ff779a84e fp 0x00007f076e87cd80 /home/ivan/lib/dart-2.10.4/bin/dart+0x1b1084e
  pc 0x0000558ff7799ab0 fp 0x00007f076e87cdd0 dart::ApiMessageWriter::WriteCObject(_Dart_CObject*)+0x1b0
  pc 0x0000558ff779aa0b fp 0x00007f076e87ce20 dart::ApiMessageWriter::WriteCMessage(_Dart_CObject*, long, dart::Message::Priority)+0x2b
  pc 0x0000558ff7d4f86a fp 0x00007f076e87cee0 Dart_PostCObject+0x3a
```

type `Dart_TypedData_kUint32` works just fine. Any plans for supporting the remaining `TypedData` types? And how can one accomplish sending Uint64Array in the meantime?

The code in dart_api_message.cc:

```c++
    case Dart_CObject_kTypedData: {
      // Write out the serialization header value for this object.
      WriteInlinedHeader(object);
      // Write out the class and tags information.
      intptr_t class_id;
      switch (object->value.as_typed_data.type) {
        case Dart_TypedData_kInt8:
          class_id = kTypedDataInt8ArrayCid;
          break;
        case Dart_TypedData_kUint8:
          class_id = kTypedDataUint8ArrayCid;
          break;
        case Dart_TypedData_kUint32:
          class_id = kTypedDataUint32ArrayCid;
          break;
        default:
          class_id = kTypedDataUint8ArrayCid;
          UNIMPLEMENTED();
      }
```

**Comment by vaind** [comment]
The workaround I'm currently using:

send the `uint64_t` array as:
```c++
object.value.as_typed_data.type = Dart_TypedData_kUint8;
object.value.as_typed_data.length = count * sizeof(uint64_t);
object.value.as_typed_data.values = count ? (uint8_t*) items : nullptr;
```

and receive in Dart with 
```dart
if (msg is! Uint8List) ... throw

Uint64List.view(msg.buffer)
```
Still it would be nice to have it implemented in dart or at least some docs (e.g. the native_api header) saying it's not available...

**Comment by rmacnak-google** [comment]
This was probably fixed by bbefc0574887c1e00e00d07d7cb8a2b5acd249d6.
